tests/api/test_shipping_method.py

```diff
@@ -305,6 +305,32 @@ def test_create_shipping_method(
     assert data["shippingZone"]["id"] == shipping_zone_id
 
 
+def test_create_shipping_method_with_invalid_price(
+    staff_api_client, shipping_zone, permission_manage_shipping,
+):
+    query = PRICE_BASED_SHIPPING_QUERY
+    name = "DHL"
+    price = -12.34
+    shipping_zone_id = graphene.Node.to_global_id("ShippingZone", shipping_zone.pk)
+    variables = {
+        "shippingZone": shipping_zone_id,
+        "name": name,
+        "price": price,
+        "minimumOrderPrice": 0,
+        "maximumOrderPrice": 20,
+        "type": ShippingMethodTypeEnum.PRICE.name,
+    }
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_shipping]
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["shippingPriceCreate"]
+    assert data["errors"][0] == {
+        "field": "price",
+        "message": "Shipping rate price cannot be lower than 0.",
```

Comment:
In the `PRICE_BASED_SHIPPING_QUERY` we should include:
```
shippingErrors {
  field
  code
}
```
And in this line instead of comparing the `message`, we should check if the `field` and `code` are as we expected.

Reply:
sure